File: app/src/main/java/com/ashiana/zlifno/alder/data/Note.java
package com.ashiana.zlifno.alder.data;

import android.arch.persistence.room.Entity;
import android.arch.persistence.room.PrimaryKey;
import android.support.annotation.NonNull;

import java.io.Serializable;

// Defines columns to add to database
@Entity(tableName = "note_table")
public class Note implements Serializable {

    public static final int NOTE_TYPE_TEXT = 1;
    public static final int NOTE_TYPE_IMAGE = 2;

    public Note(@NonNull String title, @NonNull int noteType, String content, String timeCreated) {
        this.title = title;
        this.content = content;
        this.timeCreated = timeCreated;
        this.noteType = noteType;
    }

    @PrimaryKey(autoGenerate = true)
    @NonNull
    public int id;

    @NonNull
    public String title;

    public String content;

    public String timeCreated;

    public String timeEdited;

    public int position;

    public String imagePath;

    public int noteType;

}

File: app/src/main/java/com/ashiana/zlifno/alder/data/Note.kt
package com.ashiana.zlifno.alder.data

import android.arch.persistence.room.Entity
import android.arch.persistence.room.Ignore
import android.arch.persistence.room.PrimaryKey

import java.io.Serializable
import java.util.Calendar
import java.util.Date

// Defines columns to add to database
@Entity(tableName = "note_table")
class Note(var title: String, var noteType: Int, var content: String, var timeCreated: String) : Serializable {

    @PrimaryKey(autoGenerate = true)
    var id: Int = 0

    var timeEdited: String? = null

    var position: Int = 0

    var imagePath: String? = null

    companion object {

        const val NOTE_TYPE_TEXT = 1
        const val NOTE_TYPE_IMAGE = 2
    }

}
